validator/proposer/service.go

```diff
@@ -68,10 +88,73 @@ func (p *Proposer) Stop() error {
 	return nil
 }
 
+// SaveBlockToMemory will save the block to memory so that the received
+// attestations can be aggregated in it.
+func (p *Proposer) SaveBlockToMemory(block *pbp2p.BeaconBlock) {
+	p.blockMapping[block.GetSlotNumber()] = block
+}
+
+// GetBlockFromMemory will retrieve the block to memory.
+func (p *Proposer) GetBlockFromMemory(slotnumber uint64) (*pbp2p.BeaconBlock, error) {
+	block := p.blockMapping[slotnumber]
+
+	if block == nil {
+		return nil, errors.New("block does not exist")
+	}
+
+	if block.GetSlotNumber() != slotnumber {
+		return nil, errors.New("invalid block saved in memory")
+	}
+	return block, nil
+}
+
+// DoesAttestationExist checks if an attester has already attested to a block.
+func (p *Proposer) DoesAttestationExist(attestation *pbp2p.AttestationRecord, records []*pbp2p.AttestationRecord) bool {
+	exists := false
+	for _, record := range records {
+		if bytes.Equal(record.GetAttesterBitfield(), attestation.GetAttesterBitfield()) {
+			exists = true
+			break
+		}
+	}
+	return exists
+}
+
+// AggregateAllSignatures aggregates all the signatures of the attesters. This is currently a
+// stub for now till BLS/other singature schemes are implemented.
+func (p *Proposer) AggregateAllSignatures(attestations []*pbp2p.AttestationRecord) []uint32 {
```

Comment:
`AggregateAllSignatures` and `GenerateBitmask` are seems to be out of scope for this PR. This PR is to Implement P2P messaging between validators